### ISSUE TITLE
[🔷 Feature request ]: RNNs for Dannjs

### ISSUE THREAD

**Issue by matiasvlevi** [matiasvlevi/Dann#25]
### Feature
Recurrent Neural Networks for Dannjs,

### Type
- [ ] Dann
- [ ] Matrix
- [ ] Layer
- [ ] Activation functions
- [ ] Loss functions
- [ ] Pool functions
- [ ] Datasets
- [ ] Documentation
- [ ] tests & examples
- [ ] Other 
- [x] Rann (New category for RNNs)

### Description
Adding a new class for RNNs would be interesting. The class would likely be called `Rann` for Recurrent Artificial Neural Network, instead of the `Dann` acronym which stands for Deep Artificial Neural Network. This would be a pretty time-consuming feature to implement. It would require lots of testing before we can publish the feature to the master branch. Maybe creating another testing/dev branch for this feature would be necessary.

### Examples
These are really early examples and might be completely different once we impelent the feature.
This would create the RNN.
```js
const rnn = new Rann(/*args*/);
```
We could feed series of numbers and strings, if a string value is fed we would have to bind every character to a vaslue in between 0 and 1.
```js
rnn.train([1,2,3,4,5,6,7,8]);
rnn.feed([4]);
// would return 5
```
or 
```js
rnn.train([
  'We',
  'went',
  'on',
  `a`,
  'walk'
]);
rnn.feed('a');
// would return 'walk'
```

### Note
This is, of course, early speculation about a big addition to the library. It might take quite some time to create & test, and usability might change a whole lot throughout development. I'm assigning this issue to myself because I want to start working on it, but help and criticism are welcome.

**Comment by matiasvlevi** [comment]
Added the RNN dev branch [here](https://github.com/matiasvlevi/Dann/tree/RNN)

**Comment by matiasvlevi** [comment]
Added `Rann.prototype.train` method and it seems to work well. Check it out in the [RNN branch](https://github.com/matiasvlevi/Dann/tree/RNN)
